main.go

```diff
@@ -203,19 +203,13 @@ func initFlags() {
 
 func main() {
 	var err error
-	config, err = newConfig()
+	config, err = ensureConfig()
 	if err != nil {
 		handleError(modsError{err, "Could not load your configuration file."})
 		os.Exit(1)
 	}
 
 	cache = newCache(config.CachePath)
-	if _, err := os.Stat(config.CachePath); errors.Is(err, fs.ErrNotExist) {
-		if err := os.MkdirAll(config.CachePath, 0755); err != nil {
-			handleError(modsError{err, "Could not create cache directory."})
-			os.Exit(1)
-		}
-	}
```

Comment:
Keep in mind, `config.CachePath` is not the same as config path. This would give an error creating the db if `config.CachePath` doesn't exist